control/trajectory_follower/src/pid_longitudinal_controller.cpp

```diff
@@ -448,14 +464,40 @@ LongitudinalController::Motion LongitudinalController::calcEmergencyCtrlCmd(
   const float64_t acc = trajectory_follower::longitudinal_utils::applyDiffLimitFilter(
     p.acc, m_prev_raw_ctrl_cmd.acc, dt, p.jerk);
 
-  auto clock = rclcpp::Clock{RCL_ROS_TIME};
   RCLCPP_ERROR_THROTTLE(
-    get_logger(), clock, 3000, "[Emergency stop] vel: %3.3f, acc: %3.3f", vel, acc);
+    node_->get_logger(), *node_->get_clock(), 3000, "[Emergency stop] vel: %3.3f, acc: %3.3f", vel,
+    acc);
 
   return Motion{vel, acc};
 }
 
-LongitudinalController::ControlState LongitudinalController::updateControlState(
+bool PidLongitudinalController::checkNewTrajectory()
+{
+  // flags for state transition
+  const auto & p = m_state_transition_params;
+
+  m_trajectory_buffer.push_back(*m_trajectory_ptr);
+  while (true) {
+    const auto time_diff = rclcpp::Time(m_trajectory_buffer.back().header.stamp) -
+                           rclcpp::Time(m_trajectory_buffer.front().header.stamp);
+    if (time_diff.seconds() < p.stopped_state_new_traj_duration_time) {
+      break;
+    }
+    m_trajectory_buffer.pop_front();
+  }
+
+  for (const auto & trajectory : m_trajectory_buffer) {
+    if (
+      tier4_autoware_utils::calcDistance2d(
+        trajectory.points.back().pose, m_trajectory_ptr->points.back().pose) >
+      p.stopped_state_new_traj_end_dist) {
+      return true;
+    }
+  }
+  return false;
+}
+
+PidLongitudinalController::ControlState PidLongitudinalController::updateControlState(
```

Comment:
It takes time to respond to a new trajectory, so the steer is initially in a state of convergence.

https://user-images.githubusercontent.com/39142679/175781079-a93f6175-db14-4c1e-8a8c-0668fb612c82.mp4

Reply:
( rename checkNewTrajectory -> isNewTrajectory )